src/main/java/org/scijava/log/CallingClassUtils.java

```diff
@@ -43,12 +43,44 @@ private CallingClassUtils() {
 	}
 
 	/**
+	 * Inspects the stack trace to return the name of the class that calls
+	 * this method, but ignores every class annotated with @IgnoreAsCallingClass.
+	 * <p>
+	 * If every class on the stack trace is annotated, then the class at the
+	 * root of the stack trace is returned.
+	 */
+	public static String getCallingClassName() {
+		StackTraceElement[] stackTrace = Thread.currentThread().getStackTrace();
+		for (int i = 1; i < stackTrace.length - 2; i++) {
+			String className = stackTrace[i].getClassName();
+			if (!hasIgnoreAsCallingClassAnnotation(className)) return className;
+		}
+		return stackTrace[stackTrace.length - 1].getClassName();
+	}
+
+	private static boolean hasIgnoreAsCallingClassAnnotation(String className) {
+		try {
+			Class< ? > clazz = Class.forName(className);
```

Comment:
Could you please instead use `Context.getClassLoader()`? We try to never use `Class.forName` directly throughout the SciJava codebase; the thread context class loader is used in many places throughout ImageJ, but it might return null, so you need a fallback. See cd67323c13678c931cb5e1e798149c45ef5b21a9.

Reply:
Ok, it's done